sdk/core/azure-core/azure/core/pipeline/policies/distributed_tracing.py

```diff
@@ -0,0 +1,57 @@
+from azure.core.pipeline import PipelineRequest, PipelineResponse
+from azure.core.pipeline.distributed_tracing.context import tracing_context
+from azure.core.pipeline.distributed_tracing.span import AbstractSpan, AzureSpan
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+from typing import Any, TypeVar
+
+HTTPResponseType = TypeVar("HTTPResponseType")
+HTTPRequestType = TypeVar("HTTPRequestType")
+
+
+class DistributedTracer(SansIOHTTPPolicy):
+    def __init__(
+            self,
+            name_of_spans="Azure Call",
+            header_label="span_id",
+            parent_span_param_name="parent_span",
+    ):
+        # type: (str, str, str) -> None
+        self.name_of_child_span = name_of_spans
+        self.header_label = header_label
+        self.parent_span_param_name = parent_span_param_name
+        self.span_dict = {}
+
+    def on_request(self, request, **kwargs):
+        # type: (PipelineRequest[HTTPRequestType], Any) -> None
+        parent_span = tracing_context.get_current_span()  # type: AbstractSpan
+
+        if parent_span is None:
+            parent_span = AzureSpan(name="parent_of_{}".format(self.name_of_child_span))
```

Comment:
I think we would want this to be an OC span as the default, which would mean needing a way to generate actual OC spans, which means actually importing OpenCensus at some point. It's OK to actually import OpenCensus: if tracing is enabled and the tracing backend is set to "OC". But if tracing is disabled or if the backend is not set to OC, we should not import OpenCensus.  So we could do that by: 
* having `DistributedTracer` subclasses (similar to the way transport works: https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/core/azure-core/azure/core/pipeline/transport) Users choose what backend is active by choosing which module to import, so those modules are free import concrete implementations internally, e.g. opencensus

* having one `DistributedTracer` that itself is configurable as to backend, and hides imports internally based on that configuration. 

I don't have a strong opinion on this, maybe @chlowell or @johanste can chime in

Reply:
@bryevdv, when you say "backend", what exactly do you refer to? Type of propagation, which exporter is being used or which "span type/factories" are being configured?

Either way, for opencensus, it would be greatness if the way one would ultimately enable tracing would be through a config_integration (for an example, see the [requests integration](https://github.com/census-instrumentation/opencensus-python/tree/master/contrib/opencensus-ext-requests)).

Reply:
@johanste I mean e.g. "OpenCensus" which itself is configurable along all the sub dimensions you mention. It's unclear to me how granular we want to be here. Is it enough that users can say "enable distributed tracing with OpenCensus" or do they need to be able to say "enable distributed tracing with OpenCensus and this propagator and this exporter and this context..."?

Reply:
Well if we make an extension like requests does, then we could just wrap the decorator in that extension and pass in the proper config ourselves. The client would have to do nothing but register our configuration similar to `config_integration.trace_integrations(['requests'])`. We can create parent spans for opencensus in that extension which imports Opencensus. This option requires the client to install opencensus-ext-azuresdk or whatever we name that package just like they have to install `opencensus-ext-requests` to use `config_integration.trace_integrations(['requests'])`.

This also leaves room for other implementations if the client wants as well.

Reply:
An alternate to the extension that we can do such that the user does not have to make any code changes is that the span wrappers import the implementation libraries when they are initialized. This simplifies the Policy class.

This also makes it such that we do not need the extension for all of the functionality. This way, the user would be able to enable the functionality by flipping on an environmental variable.

Reply:
This might be a bigger discussion: #5586